packages/core/src/modules/routing/protocol/pickup/v1/MessagePickupService.ts

```diff
@@ -0,0 +1,64 @@
+import type { InboundMessageContext } from '../../../../../agent/models/InboundMessageContext'
+import type { EncryptedMessage } from '../../../../../types'
+import type { BatchPickupMessage } from './messages'
+
+import { Dispatcher } from '../../../../../agent/Dispatcher'
+import { EventEmitter } from '../../../../../agent/EventEmitter'
+import { createOutboundMessage } from '../../../../../agent/helpers'
+import { InjectionSymbols } from '../../../../../constants'
+import { inject, injectable } from '../../../../../plugins'
+import { MessageRepository } from '../../../../../storage/MessageRepository'
+
+import { BatchHandler, BatchPickupHandler } from './handlers'
+import { BatchMessage, BatchMessageMessage } from './messages'
+
+@injectable()
+export class MessagePickupService {
+  private messageRepository: MessageRepository
+  private dispatcher: Dispatcher
+  private eventEmitter: EventEmitter
+
+  public constructor(
+    @inject(InjectionSymbols.MessageRepository) messageRepository: MessageRepository,
+    dispatcher: Dispatcher,
+    eventEmitter: EventEmitter
+  ) {
+    this.messageRepository = messageRepository
+    this.dispatcher = dispatcher
+    this.eventEmitter = eventEmitter
+
+    this.registerHandlers()
+  }
+
+  public async batch(messageContext: InboundMessageContext<BatchPickupMessage>) {
```

Comment:
```suggestion
  public async createBatch(messageContext: InboundMessageContext<BatchPickupMessage>) {
```

Reply:
Actually, this is mostly a relocation of the original `MessagePickUpService`, as I wanted to keep current services interfaces as most as possible in this PR.  But if this doesn't count as a 'breaking change', for sure I'll be happy to change it as suggested!

Reply:
Well, moving it to another service already is 'breaking' if we see the service as public API, or is this just a moving of the file? It's not really clear what is public / private API. I've added it to the agenda for the AFJ WG call.

Reply:
Yes, actually I meant that from outside of the framework you'd see the same MessagePickupService (i.e. when importing it from @aries-framework/core).

I agree that we should better clarify the rules for breaking changes. I don't know if I'll be able to attend today's call but my 2 cents from practical perspective is that we should care mostly on Module API and also items that are explicitly intended to be injected, such as Wallet, MessageRepository and the inbound/outbound transportes. It's desirable to do the same for services (as some custom plug-ins can make use of them), but IMHO it's acceptable if it's properly documented in the release and  it is possible to workaround any change without impacting the behaviour. For instance if we add V1 prefix to the service, compilation of the plugin using it will fail but it will not have an unpredictable behaviour.